### PR TITLE
Use kafka for watering

### PR DESCRIPTION
**Pull request by ErhardSiegl**
Solves #372

### CODE REVIEW

File: Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/WateringKafkaEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import javax.ejb.ActivationConfigProperty;
+import javax.ejb.MessageDriven;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.jms.BytesMessage;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageListener;
+// import org.jboss.ejb3.annotation.ResourceAdapter;
```
Comment (matfa42) [inline Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/WateringKafkaEndpoint.java:13]: Braucht man das Kommentar als "reminder" noch?

File: codereadyworkspace/hogarama_local.env [hunk 2]
```diff
@@ -24,3 +24,15 @@ export AMQ_URL=ssl://localhost:8883
 ##  KEYCLOAK  ##
 ################
 export KEYCLOAK_AUTH_SERVER_URL=https://keycloak-gepardec.apps.p.aws.ocp.gepardec.com/auth/
+
+################
+##  KAFKA     ##
+################
+#export KAFKA_CONNECTION=localhost:9092
+export KAFKA_CONNECTION=https://gepardec-kafka-cluster-kafka-bootstrap-kafka.apps.p.aws.ocp.gepardec.com:443
+export KAFKA_TOPIC_PREFIX="erhard."
```
Comment (matfa42) [inline codereadyworkspace/hogarama_local.env:33]: Vllt sollten wir hier einen allgemeinerern perfix fuer den master branch verwenden

File: Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/HogaramaServiceConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.gepardec.hogarama.service;
+
+public class HogaramaServiceConfiguration {
+
+    public boolean useKafkaWatering() {
```
Comment (csalmhof) [inline Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/HogaramaServiceConfiguration.java:5]: I think this methods could be static. An Injection of this class in WateringKafkaEndpoint.java and WateringMDB.java would be obsolete.

File: Hogajama/hogajama-rs/src/main/resources/META-INF/microprofile-config.properties [hunk 4]
```diff
@@ -0,0 +1,11 @@
+mp.messaging.connector.smallrye-kafka.bootstrap.servers=${kafka.bootstrap.servers}
+
+mp.messaging.incoming.habarama-in.connector=smallrye-kafka
+mp.messaging.incoming.habarama-in.topic=habarama
+mp.messaging.incoming.habarama-in.merge=true
+mp.messaging.incoming.habarama-in.broadcast=true
```
Comment (ludwigSteindl) [inline Hogajama/hogajama-rs/src/main/resources/META-INF/microprofile-config.properties:6]: The incoming stream can not be broadcasting, this statement will be ignored. Setting merge and broadcast in a single stream was introduced to the config when I defined the incoming and outgoing streams as a single connector-stream (which didn't work somehow...). I should have cleaned this up two weeks ago 👿

File: codereadyworkspace/hogarama_local.env (hunk 2 above)
Comment (ludwigSteindl) [reply codereadyworkspace/hogarama_local.env:33]: last week, we briefly talked about this issue... when multiple devs are using the same topic in AWS (at the dev-stage), they all produce to and consume from the same topic. one possible solution is a personal prefix, so each dev has its personal topic.

still, hard coding @ErhardSiegl 's personal prefix is pretty funny...

File: Hogajama/configuration/local_configuration/01_set_kafka_truststore.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+cp kafka-truststore.jks $JBOSS_HOME/standalone/configuration/
```
[the remark this answers is not in the document]
Comment (fhochleitner) [reply Hogajama/configuration/local_configuration/01_set_kafka_truststore.sh:1]: i am not sure about this. If I remember correctly this is the client certificate for accessing kafka via tls and not an authentication token.

File: Hogajama/hogajama-rs/src/main/resources/META-INF/microprofile-config.properties (hunk 4 above)
Comment (ludwigSteindl) [reply Hogajama/hogajama-rs/src/main/resources/META-INF/microprofile-config.properties:6]: In accordance to https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/model/model.html 

The Broadcast annotation changes this behavior and indicates that messages transiting in the channel are **dispatched to all the consumers**

The Merge annotation changes this behavior and indicates that a channel **can have multiple producers.**

The habarama consumer needs the enabled **merge property** if we want to be able to receive message from multiple amq topics. As of right now, we have a one-to-one connection, so we actually could remove both properties.

File: codereadyworkspace/hogarama_local.env (hunk 2 above)
Comment (ludwigSteindl) [reply codereadyworkspace/hogarama_local.env]: Sounds good to me.

File: Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/HogaramaServiceConfiguration.java (hunk 3 above)
Comment (ludwigSteindl) [reply Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/HogaramaServiceConfiguration.java:5]: IMHO, as of right now, making it static is a valid option. If we want to use other configurations, we could refactor it (yagni).

Comment (ErhardSiegl) [reply Hogajama/hogajama-rs/src/main/java/com/gepardec/hogarama/service/HogaramaServiceConfiguration.java:5]: Why refactor later when we can do the interface right the first time? A configuration object is not static in nature.

**Pull request merged**